File: Gestión_Mecanica/src/types/vehiculo.ts
/**
 * @file vehiculos.ts
 * @description Tipos e interfaces para Vehiculo - Corresponde con Vehiculo.java
 * @version 2.0
 * @date 06/03/2026
 */

import { Cliente } from "./cliente";

/**
 * Interfaz para el vehiculo
 * Corresponde exactamente con la entidad Vehiculo.java del backend.
 * Spring Boot serializa la relación ManyToOne como objeto anidado.
 */
export interface Vehiculo {
  id: number;
  marca: string;
  modelo: string;
  anio: number;
  color: string;
  patente: string;
  status: boolean;
  /** Objeto cliente serializado por Spring Boot (ManyToOne) */
  cliente?: Partial<Cliente>;
  /** Alias de conveniencia - usar cliente.id para el ID real */
  clienteId?: number;
}

export interface CreateVehiculoRequest {
  marca: string;
  modelo: string;
  anio: number;
  color: string;
  patente: string;
  status?: boolean;
  /** Spring Boot espera { "cliente": {"id": N} } para la FK */
  cliente: { id: number };
}

export interface UpdateVehiculoRequest {
  marca?: string;
  modelo?: string;
  anio?: number;
  color?: string;
  patente?: string;
  status?: boolean;
  cliente?: { id: number };
}

File: motorCloud - Backend/src/main/java/com/mangosoftware/motorCloud/Model/Entity/Vehiculo.java
package com.mangosoftware.motorCloud.Model.Entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Representa un vehículo en el sistema de gestión de taller mecánico.
 * Contiene información sobre el vehículo, como marca, modelo, año, color y
 * patente.
 * Además, mantiene una relación con el cliente propietario del vehículo.
 * La clase incluye anotaciones de JPA para mapearla a una tabla en la base de
 * datos, y utiliza Lombok para generar automáticamente los métodos getters,
 * setters,
 * constructores y otros métodos comunes.
 * 
 * @author Mangosoftware
 * @version 1.0
 */

@Entity
@Table(name = "vehiculos")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Vehiculo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String marca;

    private String modelo;

    private Integer anio;

    private String color;

    private String patente;

    @JsonIgnoreProperties({ "vehiculos", "ordenes" })
    @ManyToOne
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente; // Relación con la entidad Cliente, indicando el propietario del vehículo

    private Boolean status;
}
